arangod/Scheduler/SupervisedScheduler.cpp

```diff
@@ -108,6 +128,15 @@ bool SupervisedScheduler::queue(RequestLane lane, std::function<void()> handler)
   static thread_local uint64_t lastSubmitTime_ns;
   bool doNotify = false;
 
+
+  uint64_t approxQueueLength = _jobsSubmitted - _jobsDone;
+  if (!isDirectDeadlockLane(lane) && approxQueueLength < 10) {
```

Comment:
I think the `< 10` is valid here.
We only want the direct execution if there isn't much on the queue yet. The assumption is that if there is not much queued yet, that there are enough resources to handle requests directly, without the need to dispatch them into the queue.
The number of `10` is debatable of course, but I think the operator (<) is the correct one.

Reply:
Hm... I find this somewhat counter intuitive. Task scheduling systems usually perform spawn-to-call conversions if the task queue is full or the number of tasks reaches some threshold, i.e., if there are already more tasks than the workers can process.
I think of the scheduler as a way to schedule background tasks. Let's ignore for a second that some tasks cannot be executed directly (due to possible deadlocks). I this case no task would ever be queued, because the queue length would never grow and every queue call would result in a direct call. Is this the expected behaviour?